frontend/alert/src/App.js

```diff
@@ -131,30 +87,24 @@ const Heading = () => (
     </Center>
 );
 
-const AlertForm = ({ onSubmit, user }) => (
-    <>
-        <AutoComplete />
-        <Input placeholder="Email" value={user && user.profile.email} />
-        <Input placeholder="Phone" value={user && user.profile.phone} />
-        <Center>
-            <AlertText>
-                Alert me
-                <Dropdown>until I cancel</Dropdown>
-            </AlertText>
-            <SubmitButton onClick={onSubmit}>Submit</SubmitButton>
-        </Center>
-    </>
-);
 
-AlertForm.propTypes = {
-    onSubmit: PropTypes.func,
-    user: PropTypes.objectOf(PropTypes.any),
-};
+const genId = (() => { let counter = 0; return () => counter++; })(); // eslint-disable-line
 
 function App() {
-    const onSubmit = () => {};
     const [user, setUser] = useState(null);
-    const [page, setPage] = useState("home");
+    const [page, setPage] = useState(window.location.hash === "#manage" ? "manage" : "home");
+    const [messages, setMessages] = useState([]);
+
+    const removeMessage = k => setMessages(msgs => msgs.filter(m => m.key !== k));
+    const addMessage = ({ message, status }) => {
```

Comment:
Should we consider a setTimeout here to remove the message after x seconds?

Reply:
I was thinking about that –– wonder what @jess-tan would think